Cookbook/Technical-Documents/Architecture.md

```diff
@@ -0,0 +1,309 @@
+
+```swift
+class ViewModel: Bento.BoxViewModel {
+    let state: Property<State>
+    let routes: Signal<Route, NoError>
+
+    private let eventPipe = Feedback<Event, NoError>.input()
+
+    init(...) {
+        self.state = Property( // Defined in ReactiveFeedback
+            initial: .initial,
+            reduce: ViewModel.reduce,
+            feedbacks: [
+                actionsPipe.feedback,
+                ViewModel.whenSubmitting(...)
+            ]
+        )
+
+        self.routes = self.state
+            .filterMap { $0.route } // State to Route conversion
+    }
+
+    // User Action from UI
+    func send(action: Action) {
+        eventPipe.observer(value: .ui(action))
+    }
+
+    // MARK: - Reducer
+
+    static func reduce(_ state: State, _ event: Event) -> State {
+        switch (oldState, event) {
+        case ...:
+            return newState
+        }
+    }
+
+    // MARK: - Feedback
+
+    static func whenSubmitting(...) {
+        return Feedback(
+            predicate: { $0.status.isSubmitting },
+            effects: { state -> SignalProducer<Event, NoError> in
+                return self.validate(state: state)
+                    .flatMap(.merge) { data in
+                        self.fetchNextScreen(data)
+                    }
+                    .materializeResults()
+                    .map(Event.didFinishSubmit)
+            }
+        )
+    }
+}
+```
+
+- Input: User actions from `Renderer` will be delivered to `func send(action: Action)`, and forwarded as ReactiveFeedback's input.
+- Output: All outputs to the external are defined as `route: Signal<Route, NoError>` that triggers the next screen action observed by `FlowController`, and is (mainly) driven by `State` changes.
+    - State-driven output is so-called [Moore machine](https://en.wikipedia.org/wiki/Moore_machine)
+
+See [Babylonpartners/ReactiveFeedback](https://github.com/Babylonpartners/ReactiveFeedback) for more examples.
+
+### Renderer
+
+The minimal `Renderer` example looks as follows:
+
+```swift
+struct Renderer: BoxRenderer {
+    private let observer: Sink<Action>
+    ...
+
+    func render(state: State) -> Screen<SectionId, RowId> {
+        return Screen(
+            title: "Title",
+            box: .empty
+                |-+ renderSection0(state: state)
+                |-+ renderSection1(state: state)
+                |-? renderSection2(state: state)
+                |-? renderCurrentGP(state: state)
+        )
+    }
+
+    private func renderSection0(state: State) -> Section<SectionId, RowId> {
+        return Section<SectionId, RowId>(id: .section0)
+            |---+ Node(
+                id: .row0_0,
+                component: InputField(
+                    title: state.title,
+                    placeholder: state.placeholder,
+                    image: state.image,
+                    didTapImage: { observer(.didTapImage) },
+                    accessibilityIdentifier: String(describing: id)
```

Comment:
Since this is gonna be a reference for implementation I'd use `AccessibilityContent` here or something made up like `Accessibility.foo` or even `...`.
```suggestion
                    accessibilityIdentifier: ...
```